src/resources/2015/BarrelShrine.ts

```diff
@@ -0,0 +1,25 @@
+import { availableAmount, runChoice, visitUrl } from "kolmafia";
+import { get } from "../../property";
+import { $items } from "../../template-string";
+
+const BARRELS = $items`little firkin, normal barrel, big tun, weathered barrel, dusty barrel, disintegrating barrel, moist barrel, rotting barrel, mouldering barrel, barnacled barrel`;
+
+export function have() {
+  return get("barrelShrineUnlocked");
+}
+
+export function smashParty() {
+  if (!have()) return;
+
+  const total = BARRELS.map((i) => availableAmount(i)).reduce(
+    (sum, q) => sum + q,
+    0
+  );
```

Comment:
```suggestion
  const total = sum(BARRELS, availableAmount);
```

importing `sum` from utils. Extremely optional change.